src/main/com/algos/src/datastructures/caches/LRUCache.java

```diff
@@ -32,27 +21,27 @@ public LRUCache(int cap) {
 
     private void setCapacity(int newCapacity) {
         checkCapacity(newCapacity);
-        for (int i = data.size(); i > newCapacity; i--) {
-            Entry<K, V> evicted = evict();
-            data.remove(evicted.getKey());
-        }
         this.cap = newCapacity;
+        
     }
 
     private Entry<K, V> evict() {
         if (head == null) {
-            throw new RuntimeException("cache cannot be empty!");
+            throw new RuntimeException("Cache cannot be empty!");
```

Comment:
**Performance Issue**: Throwing RuntimeException for a cache being empty could be an overuse of exceptions which might impact performance, especially if evict is called frequently in a scenario where the cache might be empty. <br> **Fix**: Consider using a specific checked exception or returning a special value to indicate an empty cache, which could be handled more gracefully. <br> **Code Suggestion**: 
 ```
 -            throw new RuntimeException("Cache cannot be empty!");
 +            // Consider returning null or a special value instead
 ```